core/tee/tee_svc_cryp.c

```diff
@@ -40,6 +45,7 @@ struct tee_cryp_state {
 	vaddr_t key2;
 	void *ctx;
 	tee_cryp_ctx_finalize_func_t ctx_finalize;
+	enum cryp_state state;
```

Comment:
One would have thought that `bool is_initialized;` should be enough.

Reply:
Yes, as written here in this PR, that'll also work. Initially the proposal used three states (initial, update and final), but the conclusion as you maybe remember was that it was a bit overkill. Anyway, if we want to use another state in the future, then we already have the boilerplate code ready for it with this.

Reply:
I had forgot that, regardless I made the comment because I think the code would be improved if followed. But it's just a minor thing bordering on a matter of taste so it can be ignored too.